### ISSUE TITLE
Specify range for xlsx_cells()

### ISSUE THREAD

**Issue by SteveBronder** [nacnudus/tidyxl#25]
I have a large excel file that causes `xlsx_cells` to crash. It would be nice to say, "Only get this many rows and this many columns" when calling xlsx_cells.

Could something be put in [`xlsxsheet::parseSheetData`](https://github.com/nacnudus/tidyxl/blob/ae5d6e590a2ca979b69b98827a7908e4b3843817/src/xlsxsheet.cpp#L142)?

**Comment by SteveBronder** [comment]
Yes I have. My problem is that the Workbook is 67 MB in size (yes yikes!). Calling an individual sheet still causes `xlsx_cells()` to crash.

**Comment by SteveBronder** [comment]
Actually I've found that the person who made these excel files dragged the formatting down to the last possible row over a bunch of columns. So my actual issue is that for each sheet `xlsx_cells()` is trying to parse a ton of rows that only have formatting. So excel size is not really the issue, but having `n_cols` and `n_rows` arguments would be rad in solving this.

In my particular case I only need the first three rows or so.

**Comment by nacnudus** [comment]
I think a first step is to optionally omit blank cells.  When readxl implemented range import it was [complicated](https://github.com/tidyverse/readxl/pull/314), and I want to take care to do it as similarly as possible.

**Comment by SteveBronder** [comment]
That would be a nice solution for my problem!

On Fri, Apr 27, 2018 at 6:33 PM, Duncan Garmonsway <notifications@github.com

**Comment by nacnudus** [comment]
@SteveBronder blank cells can now be excluded on the master branch.

```r
xlsx_cells(x, include_blank_cells = FALSE)
```

I'll keep this issue open for the `range` feature.

**Comment by SteveBronder** [comment]
Ty so much!!
